app/models/user.rb

```diff
@@ -10,7 +10,19 @@ class User < ApplicationRecord
   pg_search_scope :search, against: :email, using: { tsearch: { prefix: true } }
 
   def gravity_user
-    @gravity_user ||= load_gravity_user
```

Comment:
I'm not understanding why we can't use this line - can we clarify why we need to use `defined?`? /cc @yuki24

Reply:
The reason is that the user may be `nil` which would pass through the `||=` test every time. It's easiest to see in the console: 
```
[1] pry(main)> @foo
=> nil
[2] pry(main)> defined?(@foo)
=> nil
[3] pry(main)> @foo = nil
=> nil
[4] pry(main)> defined?(@foo)
=> "instance-variable"
```

Reply:
Yuki did point out a couple other options for this, including the [memoist](https://github.com/matthewrudy/memoist) gem.

Reply:
Ohhhh, you're saying that the return from `load_gravity_user` could be nil and if so, then we wouldn't want to run it again, we'd want to use that nil. Hrm, that stinks, haha.

Reply:
Would be cool to use the null object pattern here but probably not worth further complicating this code. 🤷